Bot.DSharp/DSharpChannel.cs

```diff
@@ -29,15 +29,26 @@ public async Task AddOverwriteAsync(IMember m, Permissions allow, Permissions de
         {
 			if (m is DSharpMember member)
 			{
-				await Wrapped.AddOverwriteAsync(member.Wrapped, (DSharpPlus.Permissions)allow, (DSharpPlus.Permissions)deny);
+				try
+				{
+					await Wrapped.AddOverwriteAsync(member.Wrapped, (DSharpPlus.Permissions)allow, (DSharpPlus.Permissions)deny);
```

Comment:
I feel like these calls should use our DSharpPlus exception wrapper instead (which converts DSharp exceptions into Bot.Api exceptions), and then the caller should be the one handling the ignoring of the Bot.Api.UnauthorizedException.

That way:
* It's Bot.Core that is handling the feature of ignoring the Unauthorized exceptions
* That can be tested
* Other possible exceptions will become friendly (we have handling for prettifying the Bot.Api exceptions higher up).

There are 4 total instances of this pattern in this file.

Reply:
Yeah, I just forgot about that. Fixed.

Reply:
I still think the try/catch should be in the calling code, not in here, because it doesn't feel to me to be the responsibility of DSharpChannel to know what the caller wants to handle or ignore

Reply:
I see your point but right now from our bot's perspective I do kind of feel that I _don't_ want every caller to have to ignore this; it almost is a feature of our Channel that Unauthorized on channel overwrites are graceful failures instead of something to be handled